static/install-linux.sh

```diff
@@ -58,80 +58,141 @@ fi
-dest_dir_path=$PREFIX/lib/$dest_dir_name
+programs_path=$PREFIX/lib
+dest_dir_path=$programs_path/$dest_dir_name
 install_src=`pwd`
 lib_jar_name=voxelshop-start.jar
 lib_jar_path=$install_src/$lib_jar_name
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+# ^ from https://stackoverflow.com/a/246128/4541104
 
 if [ ! -f "$lib_jar_path" ]; then
-    customDie "You must run $me in same path as built $lib_jar_name."
+    customExit "You must run $me in same path as built $lib_jar_name."
 fi
 icon_img_name=voxelshop.png
 icon_img_path=$install_src/share/pixmaps/icon-48x48.png
+
+if [ ! -f "$icon_img_path" ]; then
+    try_icon_path="$SCRIPT_DIR/share/pixmaps/icon-48x48.png"
+    if [ -f "$try_icon_path" ]; then
+        echo "INFO: $icon_img_path wasn't present, so $try_icon_path will be used instead."
+        icon_img_path="$try_icon_path"
+    fi
+fi
 if [ ! -f "$icon_img_path" ]; then
     echo "WARNING: Missing '$icon_img_path' (shortcut will have no icon image)"
 fi
 shortcut_name=com.blackflux.voxelshop.desktop
 shortcut_path=$install_src/share/applications/$shortcut_name
-
+if [ ! -f "$shortcut_path" ]; then
+    try_shortcut_path="$SCRIPT_DIR/share/applications/$shortcut_name"
+    if [ -f "$try_shortcut_path" ]; then
+        echo "INFO: $shortcut_path wasn't present, so $try_shortcut_path will be used instead."
+        shortcut_path="$try_shortcut_path"
+    fi
+fi
 # endregion hardcoded settings
 
 if [ ! -d $PREFIX/lib ]; then
-    mkdir -p $PREFIX/lib || customDie "Cannot create $PREFIX/lib"
+    mkdir -p $PREFIX/lib || customExit "Cannot create $PREFIX/lib"
 fi
 
 if [ ! -d $PREFIX/bin ]; then
-    mkdir -p $PREFIX/bin || customDie "Cannot create $PREFIX/bin"
+    mkdir -p $PREFIX/bin || customExit "Cannot create $PREFIX/bin"
 fi
 
 if [ ! -d "$dest_dir_path/lib" ]; then
     msg="Cannot create '$dest_dir_path/lib'"
-    mkdir -p $dest_dir_path/lib || customDie "$msg"
+    mkdir -p $dest_dir_path/lib || customExit "$msg"
 fi
 
 if [ ! -d "$dest_dir_path/share/applications" ]; then
     msg="Cannot create '$dest_dir_path/share/applications'"
-    mkdir -p $dest_dir_path/share/applications || customDie "$msg"
+    mkdir -p $dest_dir_path/share/applications || customExit "$msg"
 fi
 install_log_name=install.log
 install_log=$dest_dir_path/$install_log_name
 if [ -f "$install_log" ]; then
     # stopping in this case avoids creating a faulty install_log that
     # may contain files the user added to the directory.
-    customDie "The program is already installed. Run uninstall-linux.sh or delete $dest_dir_path before trying to reinstall."
+    customExit "The program is already installed. Run uninstall-linux.sh or delete $dest_dir_path before trying to reinstall."
+fi
+
+if [ -d "$dest_dir_path" ]; then
+    echo "WARNING: \"$dest_dir_path\" already exists but there is no install log. The directory will be upgraded (any prepackaged files that remain will be overwritten)."
```

Comment:
Should there be a confirmation prompt here?

Reply:
I think typically you have a prompt and a `-y` flag [Reference](https://askubuntu.com/questions/672892/what-does-y-mean-in-apt-get-y-install-command), but this is fine for now. Maybe we will adjust it in the next iteration =)